components/IngredientSelector/Navigator.tsx

```diff
@@ -130,9 +134,48 @@ export default function Navigator() {
           );
         })}
       </InteractionWrapper>
+    
     );
   }
-  return (
+
+      let quantityLabel = "";
+
+      if (quantityValue !== null && quantityValue !== undefined) {
+        quantityLabel = quantityValue.toString();
+      } else {
+        quantityLabel = "";
+      }
+  
+      if (currentIngredient) {
+        const ingredient = data.ingredients[currentIngredient.ingredient_id];
+  
+        if (ingredient.category_id === "fat") {
+          if (ingredient.ingredient_name === "Huile d'olive") {
+            quantityLabel += " cuillère d'huile d'olive";
+          } else if (ingredient.ingredient_name === "Huile de colza") {
+            quantityLabel += " cuillère d'Huile de colza";
+          }
+        } else if (ingredient.category_id === "ingredient-principal") {
+          if (
+            ingredient.ingredient_name === "Poulet" ||
+            ingredient.ingredient_name === "Boeuf" ||
+            ingredient.ingredient_name === "Agneau"
+          ) {
+            quantityLabel += " gr";
+          }
+        } else if (ingredient.category_id === "vegetables") {
+          if (
+            ingredient.ingredient_name === "Oignons" ||
+            ingredient.ingredient_name === "Carottes" ||
+            ingredient.ingredient_name === "Courgettes" ||
+            ingredient.ingredient_name === "Pommes de terre"
+          ) {
+            quantityLabel = `${quantityLabel} ${ingredient.ingredient_name}`;
+          }
+        }
```

Comment:
Please don't use conditions on ids. 

You can add grams if the quantity has default_weight which means the expected value is a weight

Reply:
@alexfauquette here if i replace  ingredient.category_id by replace ingredient.category_name doesn't work